<a href="https://colab.research.google.com/github/ricardogomesmatos/MilleniumComercial/blob/main/An%C3%A1lise_SPZ.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Análise Cervejaria Speranza

Neste projeto iremos analisar os dados da cervejaria, do dia 01/01/2022 ao dia 17/04/2022. Já que a empresa vem passando por um processo de evolução, analisaremos os dados desde o início da implementação sistêmica.
Considerando que eles abrem normalmente sexta, sábado e domingo das 11h às 18h, faremos uma análise dos dias e produtos vendidos.


# Banco de Dados

Os dados fornecidos representam vendas total/dia por item. 

In [1]:
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt


In [2]:
tabela_vendas = pd.read_excel('Analise_2022_SPZ.xlsx')
tabela_vendas

Cód produto                 Descrição  Quantidade Vendida  \
0              72                      AGUA                  11   
1              73              AGUA COM GAS                   6   
2             362  BATATA FRITA TRADICIONAL                   4   
3              42            BATATA RUSTICA                   5   
4              83       BOLINHO DE BACALHAU                   3   
...           ...                       ...                 ...   
1436           48               STEAK ANCHO                   4   
1437          231                   TILÁPIA                  12   
1438           68                 UVA 300ML                   6   
1439           70                 UVA 900ML                   1   
1440           63              VIENNA LAGER                  21   

      Valor Unitário  Valor Bruto       Data  
0                4.0         44.0 2022-01-01  
1                5.0         30.0 2022-01-01  
2               22.0         88.0 2022-01-01  
3               28.0        140.0 2022-01-01  
4               40.0        120.0 2022-01-01  
...              ...          ...        ...  
1436           100.0        400.0 2022-04-17  
1437            55.0        660.0 2022-04-17  
1438             6.0         36.0 2022-04-17  
1439            13.0         13.0 2022-04-17  
1440            13.0        273.0 2022-04-17  

[1441 rows x 6 columns]

In [3]:
renomear = {'Cód produto' : 'cod',
            'Descrição' : 'descricao',
            'Quantidade Vendida' : 'quantidade_vendida',
            'Valor Unitário' : 'valor_unitario',
            'Valor Bruto' : 'valor_bruto',
            'Data' : 'data'}
tabela_vendas = tabela_vendas.rename(columns = renomear)
tabela_vendas.head(5)

cod                 descricao  quantidade_vendida  valor_unitario  \
0   72                      AGUA                  11             4.0   
1   73              AGUA COM GAS                   6             5.0   
2  362  BATATA FRITA TRADICIONAL                   4            22.0   
3   42            BATATA RUSTICA                   5            28.0   
4   83       BOLINHO DE BACALHAU                   3            40.0   

   valor_bruto       data  
0         44.0 2022-01-01  
1         30.0 2022-01-01  
2         88.0 2022-01-01  
3        140.0 2022-01-01  
4        120.0 2022-01-01

In [4]:
item_vendido = tabela_vendas[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto']].groupby(['descricao', 'cod']).sum()
item_vendido

quantidade_vendida  valor_bruto
descricao                          cod                                 
ADICIONAL PAO                      59                   16         94.5
AGUA                               72                  497       1988.0
AGUA COM GAS                       73                  295       1475.0
ARROZ                              53                  193       1158.0
ARROZ CARRETEIRO                   396                 114       1815.0
...                                                    ...          ...
UVA 900ML                          70                   68        884.0
VIENNA LAGER                       63                  648       8530.0
VIENNA LITRO/ GROWLER              326                   3        102.0
VINHO CASILLERO DEL DIABLO GARRAFA 298                   3        240.0
VINHO SANTA HELENA GARRAFA         297                   1         60.0

[72 rows x 2 columns]

In [5]:
tabela_vendas

cod                 descricao  quantidade_vendida  valor_unitario  \
0      72                      AGUA                  11             4.0   
1      73              AGUA COM GAS                   6             5.0   
2     362  BATATA FRITA TRADICIONAL                   4            22.0   
3      42            BATATA RUSTICA                   5            28.0   
4      83       BOLINHO DE BACALHAU                   3            40.0   
...   ...                       ...                 ...             ...   
1436   48               STEAK ANCHO                   4           100.0   
1437  231                   TILÁPIA                  12            55.0   
1438   68                 UVA 300ML                   6             6.0   
1439   70                 UVA 900ML                   1            13.0   
1440   63              VIENNA LAGER                  21            13.0   

      valor_bruto       data  
0            44.0 2022-01-01  
1            30.0 2022-01-01  
2            88.0 2022-01-01  
3           140.0 2022-01-01  
4           120.0 2022-01-01  
...           ...        ...  
1436        400.0 2022-04-17  
1437        660.0 2022-04-17  
1438         36.0 2022-04-17  
1439         13.0 2022-04-17  
1440        273.0 2022-04-17  

[1441 rows x 6 columns]

#Classificação dos itens

A classificação dos tipos servirá para fazer um estudo separado e descritivo de cada produto, separado por segmento. Servirá para uso no dashboard no PowerBI.

In [6]:
list(tabela_vendas['descricao'].drop_duplicates())

['AGUA',
 'AGUA COM GAS',
 'BATATA FRITA TRADICIONAL',
 'BATATA RUSTICA',
 'BOLINHO DE BACALHAU',
 'CACHAÇA',
 'CHOPP VINHO',
 'COCADA DE FORNO',
 'COCA LATA',
 'CONSERVA SPERANZA',
 'CORDEIRO PERNIL',
 'COSTELA DE PORCO DEFUMADA',
 'DADINHO DE TAPIOCA',
 'FRITAS TRADICIONAL',
 'GUARANA LATA',
 'HEINEKEN SEM ALCOOL',
 'IPA',
 'LARANJA 300ML',
 'LARANJA 900ML',
 'LINGUIÇA',
 'PICANHA',
 'PILSEN',
 'PORCO NA LATA',
 "PRATO KID'S",
 'RAGÚ DE CUPIM',
 'REGUA DEGUSTACAO',
 'STEAK ANCHO',
 'TAÇA DE ESPUMANTE',
 'TILÁPIA',
 'UVA 300ML',
 'VIENNA LAGER',
 'ADICIONAL PAO',
 'ARROZ',
 'CONSERVA GILO',
 'FAROFA BACON',
 'FEIJÃO',
 'GIN',
 'IPA LITRO/ GROWLER',
 'ISCA CARNE SOL',
 'MEIA PORÇÃO DE PICANHA',
 'MOLHO CHIMICHURI',
 'MOLHO MAIONESE',
 'PAIOL UNIDADE',
 'PURÊ DE BANANA DA TERRA',
 'UVA 900ML',
 'CONSERVAS PEPINO',
 'MEIA PORÇÃO DE PORCO NA LATA',
 'MOLHO MOSTARDA DIJON',
 'TORTA CRUMBLE DE MORANGO',
 'VINHO CASILLERO DEL DIABLO GARRAFA',
 'MEIA PORÇÃO DE FRITAS',
 'MEIA PORÇÃO DE ISCA D

## Cervejas Especiais

In [7]:
cervas_copo = ['CHOPP VINHO',
'IPA',
'PILSEN',
'REGUA DEGUSTACAO',
'VIENNA LAGER',
'GIN',
'BELGA',
 ]


In [8]:
especiais = tabela_vendas['descricao'].isin(cervas_copo)

In [9]:
dados_cerva = tabela_vendas[especiais]
dados_cerva

cod         descricao  quantidade_vendida  valor_unitario  valor_bruto  \
6      64       CHOPP VINHO                  13            12.0        156.0   
16     62               IPA                  46            14.0        644.0   
21     60            PILSEN                  52            11.0        572.0   
25     57  REGUA DEGUSTACAO                  14            28.0        392.0   
30     63      VIENNA LAGER                  41            13.0        533.0   
...   ...               ...                 ...             ...          ...   
1420   65               GIN                   1            25.0         25.0   
1422   62               IPA                  21            14.0        294.0   
1429   60            PILSEN                  83            11.0        913.0   
1435   57  REGUA DEGUSTACAO                   8            28.0        224.0   
1440   63      VIENNA LAGER                  21            13.0        273.0   

           data  
6    2022-01-01  
16   2022-01-01  
21   2022-01-01  
25   2022-01-01  
30   2022-01-01  
...         ...  
1420 2022-04-17  
1422 2022-04-17  
1429 2022-04-17  
1435 2022-04-17  
1440 2022-04-17  

[288 rows x 6 columns]

In [10]:
dados_cerva.index = range(dados_cerva.shape[0])

In [11]:
dados_cerva

cod         descricao  quantidade_vendida  valor_unitario  valor_bruto  \
0     64       CHOPP VINHO                  13            12.0        156.0   
1     62               IPA                  46            14.0        644.0   
2     60            PILSEN                  52            11.0        572.0   
3     57  REGUA DEGUSTACAO                  14            28.0        392.0   
4     63      VIENNA LAGER                  41            13.0        533.0   
..   ...               ...                 ...             ...          ...   
283   65               GIN                   1            25.0         25.0   
284   62               IPA                  21            14.0        294.0   
285   60            PILSEN                  83            11.0        913.0   
286   57  REGUA DEGUSTACAO                   8            28.0        224.0   
287   63      VIENNA LAGER                  21            13.0        273.0   

          data  
0   2022-01-01  
1   2022-01-01  
2   2022-01-01  
3   2022-01-01  
4   2022-01-01  
..         ...  
283 2022-04-17  
284 2022-04-17  
285 2022-04-17  
286 2022-04-17  
287 2022-04-17  

[288 rows x 6 columns]

In [12]:
dados_cerva.to_excel('cervejas_especiais.xlsx', index=True)

In [13]:
especiais = dados_cerva[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto', 'data']].groupby(['cod', 'descricao', 'data']).sum()
especiais

quantidade_vendida  valor_bruto
cod descricao        data                                       
57  REGUA DEGUSTACAO 2022-01-01                  14        392.0
                     2022-01-02                  16        448.0
                     2022-01-07                   6        168.0
                     2022-01-08                   7        196.0
                     2022-01-09                   7        196.0
...                                             ...          ...
206 BELGA            2022-04-08                   2         32.0
                     2022-04-09                  12        192.0
                     2022-04-10                   9        144.0
                     2022-04-16                  16        256.0
                     2022-04-17                  11        176.0

[288 rows x 2 columns]

## Pratos

In [14]:
pratos = [ 'BATATA FRITA TRADICIONAL',
 'BATATA RUSTICA',
 'BOLINHO DE BACALHAU',
 'CONSERVA SPERANZA',
 'CORDEIRO PERNIL',
 'COSTELA DE PORCO DEFUMADA',
 'DADINHO DE TAPIOCA',
 'FRITAS TRADICIONAL',
 'LINGUIÇA',
 'PICANHA',
 'PORCO NA LATA',
 "PRATO KID'S",
 'RAGÚ DE CUPIM',
 'STEAK ANCHO',
 'TILÁPIA',
 'ADICIONAL PAO',
 'ARROZ',
 'CONSERVA GILO',
 'FAROFA BACON',
 'FEIJÃO',
 'ISCA CARNE SOL',
 'MEIA PORÇÃO DE PICANHA',
 'PURÊ DE BANANA DA TERRA',
 'CONSERVAS PEPINO',
 'MEIA PORÇÃO DE PORCO NA LATA',
 'MEIA PORÇÃO DE FRITAS',
 'MEIA PORÇÃO DE ISCA DE CARNE DE SOL',
 'CONSERVA CEBOLA',
 'ARROZ CARRETEIRO',
 'EVENTO BATATA FRITA',
 'EVENTO CHOPP VERDE',
 'EVENTO ISCA DE BOI',
 'EVENTO ISCA DE FRANGO',
 'EVENTO ISCA DE PORCO',
 'MEIA PORÇÃO DE LINGUIÇA',
 'REFEIÇÃO']

In [15]:
refeicao = tabela_vendas['descricao'].isin(pratos)
dados_pratos = tabela_vendas[refeicao]
dados_pratos

cod                 descricao  quantidade_vendida  valor_unitario  \
2     362  BATATA FRITA TRADICIONAL                   4            22.0   
3      42            BATATA RUSTICA                   5            28.0   
4      83       BOLINHO DE BACALHAU                   3            40.0   
9      40         CONSERVA SPERANZA                   1            26.0   
10     82           CORDEIRO PERNIL                   1            90.0   
...   ...                       ...                 ...             ...   
1431  212               PRATO KID'S                   4            30.0   
1433  334   PURÊ DE BANANA DA TERRA                   1            10.0   
1434  403                  REFEIÇÃO                  13            35.0   
1436   48               STEAK ANCHO                   4           100.0   
1437  231                   TILÁPIA                  12            55.0   

      valor_bruto       data  
2            88.0 2022-01-01  
3           140.0 2022-01-01  
4           120.0 2022-01-01  
9            26.0 2022-01-01  
10           90.0 2022-01-01  
...           ...        ...  
1431        120.0 2022-04-17  
1433         10.0 2022-04-17  
1434        455.0 2022-04-17  
1436        400.0 2022-04-17  
1437        660.0 2022-04-17  

[620 rows x 6 columns]

In [16]:
dados_pratos.index = range(dados_pratos.shape[0])
dados_pratos

cod                 descricao  quantidade_vendida  valor_unitario  \
0    362  BATATA FRITA TRADICIONAL                   4            22.0   
1     42            BATATA RUSTICA                   5            28.0   
2     83       BOLINHO DE BACALHAU                   3            40.0   
3     40         CONSERVA SPERANZA                   1            26.0   
4     82           CORDEIRO PERNIL                   1            90.0   
..   ...                       ...                 ...             ...   
615  212               PRATO KID'S                   4            30.0   
616  334   PURÊ DE BANANA DA TERRA                   1            10.0   
617  403                  REFEIÇÃO                  13            35.0   
618   48               STEAK ANCHO                   4           100.0   
619  231                   TILÁPIA                  12            55.0   

     valor_bruto       data  
0           88.0 2022-01-01  
1          140.0 2022-01-01  
2          120.0 2022-01-01  
3           26.0 2022-01-01  
4           90.0 2022-01-01  
..           ...        ...  
615        120.0 2022-04-17  
616         10.0 2022-04-17  
617        455.0 2022-04-17  
618        400.0 2022-04-17  
619        660.0 2022-04-17  

[620 rows x 6 columns]

In [17]:
dados_pratos.to_excel('dados_pratos.xlsx', index=True)

In [18]:
pratosespeciais = dados_pratos[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto', 'data']].groupby(['cod', 'descricao', 'data']).sum()
pratosespeciais.head(15)

quantidade_vendida  valor_bruto
cod descricao         data                                       
40  CONSERVA SPERANZA 2022-01-01                   1         26.0
                      2022-01-02                   2         52.0
                      2022-01-07                   3         78.0
                      2022-01-08                   3         78.0
                      2022-01-13                   1         26.0
                      2022-01-16                   3         78.0
                      2022-01-23                   3         78.0
                      2022-01-30                   1         26.0
                      2022-02-04                   1         26.0
                      2022-02-06                   1         26.0
                      2022-02-13                   6        156.0
                      2022-02-19                   5        130.0
                      2022-02-26                   4        104.0
                      2022-02-27                   4        104.0
                      2022-02-28                   2         52.0

## Bebidas não especiais

In [19]:
bebidas = ['AGUA',
 'AGUA COM GAS',
 'CACHAÇA',
 'COCA LATA',
 'GUARANA LATA',
 'HEINEKEN SEM ALCOOL',
 'LARANJA 300ML',
 'LARANJA 900ML',
 'TAÇA DE ESPUMANTE',
 'UVA 300ML',
 'UVA 900ML',
 'VINHO CASILLERO DEL DIABLO GARRAFA',
 'UVA',
 'VINHO SANTA HELENA GARRAFA',
 ]


In [20]:
bebidasextras = tabela_vendas['descricao'].isin(bebidas)
dados_bebida = tabela_vendas[bebidasextras]
dados_bebida

cod      descricao  quantidade_vendida  valor_unitario  valor_bruto  \
0      72           AGUA                  11             4.0         44.0   
1      73   AGUA COM GAS                   6             5.0         30.0   
5      75        CACHAÇA                   1             6.0          6.0   
8      66      COCA LATA                   5             6.0         30.0   
14     67   GUARANA LATA                   4             6.0         24.0   
...   ...            ...                 ...             ...          ...   
1415   66      COCA LATA                  18             6.0        108.0   
1421   67   GUARANA LATA                   9             6.0         54.0   
1424   69  LARANJA 300ML                   7             6.0         42.0   
1438   68      UVA 300ML                   6             6.0         36.0   
1439   70      UVA 900ML                   1            13.0         13.0   

           data  
0    2022-01-01  
1    2022-01-01  
5    2022-01-01  
8    2022-01-01  
14   2022-01-01  
...         ...  
1415 2022-04-17  
1421 2022-04-17  
1424 2022-04-17  
1438 2022-04-17  
1439 2022-04-17  

[393 rows x 6 columns]

In [21]:
dados_bebida.index = range(dados_bebida.shape[0])
dados_bebida

cod      descricao  quantidade_vendida  valor_unitario  valor_bruto  \
0     72           AGUA                  11             4.0         44.0   
1     73   AGUA COM GAS                   6             5.0         30.0   
2     75        CACHAÇA                   1             6.0          6.0   
3     66      COCA LATA                   5             6.0         30.0   
4     67   GUARANA LATA                   4             6.0         24.0   
..   ...            ...                 ...             ...          ...   
388   66      COCA LATA                  18             6.0        108.0   
389   67   GUARANA LATA                   9             6.0         54.0   
390   69  LARANJA 300ML                   7             6.0         42.0   
391   68      UVA 300ML                   6             6.0         36.0   
392   70      UVA 900ML                   1            13.0         13.0   

          data  
0   2022-01-01  
1   2022-01-01  
2   2022-01-01  
3   2022-01-01  
4   2022-01-01  
..         ...  
388 2022-04-17  
389 2022-04-17  
390 2022-04-17  
391 2022-04-17  
392 2022-04-17  

[393 rows x 6 columns]

In [22]:
dados_bebida.to_excel('bebidas_extras.xlsx', index=True)

In [23]:
tabela_bebidas_extras = dados_bebida[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto', 'data']].groupby(['cod', 'descricao', 'data']).sum()
tabela_bebidas_extras

quantidade_vendida  valor_bruto
cod descricao         data                                       
66  COCA LATA         2022-01-01                   5         30.0
                      2022-01-02                  16         96.0
                      2022-01-07                   3         18.0
                      2022-01-08                  13         78.0
                      2022-01-13                   2         12.0
...                                              ...          ...
361 TAÇA DE ESPUMANTE 2022-04-02                   2         20.0
                      2022-04-03                   2         20.0
                      2022-04-09                   3         30.0
                      2022-04-10                   1         10.0
                      2022-04-16                   3         30.0

[393 rows x 2 columns]

## Sobremesas

In [24]:
sobremesas = ['COCADA DE FORNO',
 'TORTA CRUMBLE DE MORANGO',
 'PUDIM TRADICIONAL',
 ]

In [25]:
sobremesas_extra = tabela_vendas['descricao'].isin(sobremesas)
sobremesas_extra = tabela_vendas[sobremesas_extra]
sobremesas_extra

cod                 descricao  quantidade_vendida  valor_unitario  \
7     355           COCADA DE FORNO                  13            18.0   
40    355           COCADA DE FORNO                  12            18.0   
101   355           COCADA DE FORNO                   7            18.0   
132   355           COCADA DE FORNO                   8            18.0   
161   370  TORTA CRUMBLE DE MORANGO                   1            18.0   
172   355           COCADA DE FORNO                   3            18.0   
195   370  TORTA CRUMBLE DE MORANGO                   1            18.0   
220   355           COCADA DE FORNO                   4            18.0   
248   355           COCADA DE FORNO                   6            18.0   
282   355           COCADA DE FORNO                   4            18.0   
334   355           COCADA DE FORNO                   6            18.0   
371   355           COCADA DE FORNO                   7            18.0   
408   355           COCADA DE FORNO                   6            18.0   
426   355           COCADA DE FORNO                   9            18.0   
456   355           COCADA DE FORNO                   5            18.0   
496   355           COCADA DE FORNO                  10            18.0   
527   355           COCADA DE FORNO                   9            18.0   
570   355           COCADA DE FORNO                  10            18.0   
609   355           COCADA DE FORNO                  11            18.0   
659   355           COCADA DE FORNO                   6            18.0   
689   355           COCADA DE FORNO                   7            18.0   
722   355           COCADA DE FORNO                   2            18.0   
744   355           COCADA DE FORNO                   9            18.0   
782   355           COCADA DE FORNO                  12            18.0   
805   370  TORTA CRUMBLE DE MORANGO                   2            18.0   
820   355           COCADA DE FORNO                   5            18.0   
861   355           COCADA DE FORNO                   6            18.0   
899   355           COCADA DE FORNO                   2            18.0   
928   355           COCADA DE FORNO                   8            18.0   
949    55         PUDIM TRADICIONAL                   1            10.0   
965   355           COCADA DE FORNO                   3            18.0   
984    55         PUDIM TRADICIONAL                   1            10.0   
1015  355           COCADA DE FORNO                   3            18.0   
1031   55         PUDIM TRADICIONAL                   3            10.0   
1047  355           COCADA DE FORNO                   7            18.0   
1064   55         PUDIM TRADICIONAL                   2            10.0   
1128  355           COCADA DE FORNO                   1            18.0   
1160  355           COCADA DE FORNO                   2            18.0   
1183   55         PUDIM TRADICIONAL                   2            10.0   
1197  355           COCADA DE FORNO                   1            18.0   
1217  355           COCADA DE FORNO                   8            18.0   
1234   55         PUDIM TRADICIONAL                   4            10.0   
1248  355           COCADA DE FORNO                   4            18.0   
1268   55         PUDIM TRADICIONAL                   4            10.0   
1298  355           COCADA DE FORNO                   6            18.0   
1316   55         PUDIM TRADICIONAL                   1            10.0   
1333  355           COCADA DE FORNO                   7            18.0   
1356   55         PUDIM TRADICIONAL                   2            10.0   
1373  355           COCADA DE FORNO                   2            18.0   
1397   55         PUDIM TRADICIONAL                   1            10.0   
1414  355           COCADA DE FORNO                   7            18.0   
1432   55         PUDIM TRADICIONAL                   1            10.0   

      valor_bruto       data  

In [26]:
sobremesas_extra.index = range(sobremesas_extra.shape[0])
sobremesas_extra

cod                 descricao  quantidade_vendida  valor_unitario  \
0   355           COCADA DE FORNO                  13            18.0   
1   355           COCADA DE FORNO                  12            18.0   
2   355           COCADA DE FORNO                   7            18.0   
3   355           COCADA DE FORNO                   8            18.0   
4   370  TORTA CRUMBLE DE MORANGO                   1            18.0   
5   355           COCADA DE FORNO                   3            18.0   
6   370  TORTA CRUMBLE DE MORANGO                   1            18.0   
7   355           COCADA DE FORNO                   4            18.0   
8   355           COCADA DE FORNO                   6            18.0   
9   355           COCADA DE FORNO                   4            18.0   
10  355           COCADA DE FORNO                   6            18.0   
11  355           COCADA DE FORNO                   7            18.0   
12  355           COCADA DE FORNO                   6            18.0   
13  355           COCADA DE FORNO                   9            18.0   
14  355           COCADA DE FORNO                   5            18.0   
15  355           COCADA DE FORNO                  10            18.0   
16  355           COCADA DE FORNO                   9            18.0   
17  355           COCADA DE FORNO                  10            18.0   
18  355           COCADA DE FORNO                  11            18.0   
19  355           COCADA DE FORNO                   6            18.0   
20  355           COCADA DE FORNO                   7            18.0   
21  355           COCADA DE FORNO                   2            18.0   
22  355           COCADA DE FORNO                   9            18.0   
23  355           COCADA DE FORNO                  12            18.0   
24  370  TORTA CRUMBLE DE MORANGO                   2            18.0   
25  355           COCADA DE FORNO                   5            18.0   
26  355           COCADA DE FORNO                   6            18.0   
27  355           COCADA DE FORNO                   2            18.0   
28  355           COCADA DE FORNO                   8            18.0   
29   55         PUDIM TRADICIONAL                   1            10.0   
30  355           COCADA DE FORNO                   3            18.0   
31   55         PUDIM TRADICIONAL                   1            10.0   
32  355           COCADA DE FORNO                   3            18.0   
33   55         PUDIM TRADICIONAL                   3            10.0   
34  355           COCADA DE FORNO                   7            18.0   
35   55         PUDIM TRADICIONAL                   2            10.0   
36  355           COCADA DE FORNO                   1            18.0   
37  355           COCADA DE FORNO                   2            18.0   
38   55         PUDIM TRADICIONAL                   2            10.0   
39  355           COCADA DE FORNO                   1            18.0   
40  355           COCADA DE FORNO                   8            18.0   
41   55         PUDIM TRADICIONAL                   4            10.0   
42  355           COCADA DE FORNO                   4            18.0   
43   55         PUDIM TRADICIONAL                   4            10.0   
44  355           COCADA DE FORNO                   6            18.0   
45   55         PUDIM TRADICIONAL                   1            10.0   
46  355           COCADA DE FORNO                   7            18.0   
47   55         PUDIM TRADICIONAL                   2            10.0   
48  355           COCADA DE FORNO                   2            18.0   
49   55         PUDIM TRADICIONAL                   1            10.0   
50  355           COCADA DE FORNO                   7            18.0   
51   55         PUDIM TRADICIONAL                   1            10.0   

    valor_bruto       data  
0         234.0 2022-01-01  
1         216.0 2022-01-02  
2         126.0 2022-01-07  
3         144.0 20

In [27]:
sobremesas_extra.to_excel('sobremesas_extra.xlsx', index=True)

In [28]:
tabela_sobremesas_extra = sobremesas_extra[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto', 'data']].groupby(['cod', 'descricao', 'data']).sum()
tabela_sobremesas_extra

quantidade_vendida  valor_bruto
cod descricao                data                                       
55  PUDIM TRADICIONAL        2022-03-05                   1         10.0
                             2022-03-06                   1         10.0
                             2022-03-12                   3         30.0
                             2022-03-13                   2         20.0
                             2022-03-27                   2         20.0
                             2022-04-02                   4         40.0
                             2022-04-03                   4         40.0
                             2022-04-09                   1         10.0
                             2022-04-10                   2         20.0
                             2022-04-16                   1         10.0
                             2022-04-17                   1         10.0
355 COCADA DE FORNO          2022-01-01                  13        234.0
                             2022-01-02                  12        216.0
                             2022-01-07                   7        126.0
                             2022-01-08                   8        144.0
                             2022-01-09                   3         54.0
                             2022-01-14                   4         72.0
                             2022-01-15                   6        108.0
                             2022-01-16                   4         72.0
                             2022-01-22                   6        108.0
                             2022-01-23                   7        126.0
                             2022-01-28                   6        108.0
                             2022-01-29                   9        162.0
                             2022-01-30                   5         90.0
                             2022-02-05                  10        180.0
                             2022-02-06                   9        162.0
                             2022-02-12                  10        180.0
                             2022-02-13                  11        198.0
                             2022-02-19                   6        108.0
                             2022-02-20                   7        126.0
                             2022-02-25                   2         36.0
                             2022-02-26                   9        162.0
                             2022-02-27                  12        216.0
                             2022-02-28                   5         90.0
                             2022-03-01                   6        108.0
                             2022-03-04                   2         36.0
                             2022-03-05                   8        144.0
                             2022-03-06                   3         54.0
                             2022-03-12                   3         54.0
                             2022-03-13                   7        126.0
                             2022-03-26                   1         18.0
                             2022-03-27                   2         36.0
                             2022-04-01                   1         18.0
                             2022-04-02                   8        144.0
                             2022-04-03                   4         72.0
                             2022-04-09                   6        108.0
                             2022-04-10                   7        126.0
                             2022-04-16                   2         36.0
                             2022-04-17                   7        126.0
370 TORTA CRUMBLE DE MORANGO 2022-01-08                   1         18.0
                             2022-01-09                   1         18.0
                             2022-02-27                   2         36.0

# Tabela segmentada
Para termos um melhor entendimento dos gráficos que desenvolveremos futuramente, segmentar a tabela por tipo fará com que, com uma visão geral, possamos focar numa análise de dados específica.

In [29]:
tabela_segmentada = tabela_vendas.copy()

In [30]:
renomearcomida = { 'BATATA FRITA TRADICIONAL': 'alimento',
 'BATATA RUSTICA': 'alimento',
 'BOLINHO DE BACALHAU': 'alimento',
 'CONSERVA SPERANZA': 'alimento',
 'CORDEIRO PERNIL': 'alimento',
 'COSTELA DE PORCO DEFUMADA': 'alimento',
 'DADINHO DE TAPIOCA': 'alimento',
 'FRITAS TRADICIONAL': 'alimento',
 'LINGUIÇA': 'alimento',
 'PICANHA': 'alimento',
 'PORCO NA LATA': 'alimento',
 "PRATO KID'S": 'alimento',
 'RAGÚ DE CUPIM': 'alimento',
 'STEAK ANCHO': 'alimento',
 'TILÁPIA': 'alimento',
 'ADICIONAL PAO': 'alimento',
 'ARROZ': 'alimento',
 'CONSERVA GILO': 'alimento',
 'FAROFA BACON': 'alimento',
 'FEIJÃO': 'alimento',
 'ISCA CARNE SOL': 'alimento',
 'MEIA PORÇÃO DE PICANHA': 'alimento',
 'PURÊ DE BANANA DA TERRA': 'alimento',
 'CONSERVAS PEPINO': 'alimento',
 'MEIA PORÇÃO DE PORCO NA LATA': 'alimento',
 'MEIA PORÇÃO DE FRITAS': 'alimento',
 'MEIA PORÇÃO DE ISCA DE CARNE DE SOL': 'alimento',
 'CONSERVA CEBOLA': 'alimento',
 'ARROZ CARRETEIRO': 'alimento',
 'EVENTO BATATA FRITA': 'alimento',
 'EVENTO CHOPP VERDE': 'alimento',
 'EVENTO ISCA DE BOI': 'alimento',
 'EVENTO ISCA DE FRANGO': 'alimento',
 'EVENTO ISCA DE PORCO': 'alimento',
 'MEIA PORÇÃO DE LINGUIÇA': 'alimento',
 'REFEIÇÃO': 'alimento'}

In [31]:
tabela_segmentada['descricao'].value_counts()
tabela_segmentada.replace({ 'BATATA FRITA TRADICIONAL': 'alimento',
 'BATATA RUSTICA': 'alimento',
 'BOLINHO DE BACALHAU': 'alimento',
 'CONSERVA SPERANZA': 'alimento',
 'CORDEIRO PERNIL': 'alimento',
 'COSTELA DE PORCO DEFUMADA': 'alimento',
 'DADINHO DE TAPIOCA': 'alimento',
 'FRITAS TRADICIONAL': 'alimento',
 'LINGUIÇA': 'alimento',
 'PICANHA': 'alimento',
 'PORCO NA LATA': 'alimento',
 "PRATO KID'S": 'alimento',
 'RAGÚ DE CUPIM': 'alimento',
 'STEAK ANCHO': 'alimento',
 'TILÁPIA': 'alimento',
 'ADICIONAL PAO': 'alimento',
 'ARROZ': 'alimento',
 'CONSERVA GILO': 'alimento',
 'FAROFA BACON': 'alimento',
 'FEIJÃO': 'alimento',
 'ISCA CARNE SOL': 'alimento',
 'MEIA PORÇÃO DE PICANHA': 'alimento',
 'PURÊ DE BANANA DA TERRA': 'alimento',
 'CONSERVAS PEPINO': 'alimento',
 'MEIA PORÇÃO DE PORCO NA LATA': 'alimento',
 'MEIA PORÇÃO DE FRITAS': 'alimento',
 'MEIA PORÇÃO DE ISCA DE CARNE DE SOL': 'alimento',
 'CONSERVA CEBOLA': 'alimento',
 'ARROZ CARRETEIRO': 'alimento',
 'EVENTO BATATA FRITA': 'alimento',
 'EVENTO CHOPP VERDE': 'alimento',
 'EVENTO ISCA DE BOI': 'alimento',
 'EVENTO ISCA DE FRANGO': 'alimento',
 'EVENTO ISCA DE PORCO': 'alimento',
 'MEIA PORÇÃO DE LINGUIÇA': 'alimento',
 'REFEIÇÃO': 'alimento',
 'MOLHO MAIONESE': 'alimento',
 'MOLHO PIMENTAO VERMELHO': 'alimento',
 'MOLHO CHIMICHURI': 'alimento', 
 'MOLHO MOSTARDA DIJON': 'alimento' }, inplace=True)
tabela_segmentada
tabela_segmentada['descricao'].value_counts()

alimento                              666
PILSEN                                 50
IPA                                    47
AGUA                                   45
AGUA COM GAS                           44
VIENNA LAGER                           43
CHOPP VINHO                            43
COCA LATA                              41
REGUA DEGUSTACAO                       41
GUARANA LATA                           40
UVA 300ML                              39
COCADA DE FORNO                        38
LARANJA 300ML                          35
TAÇA DE ESPUMANTE                      33
BELGA                                  33
LARANJA 900ML                          32
GIN                                    31
UVA 900ML                              27
HEINEKEN SEM ALCOOL                    26
CACHAÇA                                25
PAIOL UNIDADE                          12
PUDIM TRADICIONAL                      11
IPA LITRO/ GROWLER                      8
PILSEN LITRO/ GROWLER             

In [32]:
tabela_segmentada['descricao'].value_counts()
tabela_segmentada.replace({'CHOPP VINHO': 'chopp',
'IPA': 'chopp',
'PILSEN': 'chopp',
'REGUA DEGUSTACAO': 'chopp',
'VIENNA LAGER': 'chopp',
'GIN': 'chopp',
'BELGA': 'chopp'}, inplace=True)
tabela_segmentada
tabela_segmentada['descricao'].value_counts()

alimento                              666
chopp                                 288
AGUA                                   45
AGUA COM GAS                           44
COCA LATA                              41
GUARANA LATA                           40
UVA 300ML                              39
COCADA DE FORNO                        38
LARANJA 300ML                          35
TAÇA DE ESPUMANTE                      33
LARANJA 900ML                          32
UVA 900ML                              27
HEINEKEN SEM ALCOOL                    26
CACHAÇA                                25
PAIOL UNIDADE                          12
PUDIM TRADICIONAL                      11
IPA LITRO/ GROWLER                      8
PILSEN LITRO/ GROWLER                   6
CHOPP VINHO LITRO/ GROWLER              6
BELGA LITRO/ GROWLER                    6
UVA                                     3
TORTA CRUMBLE DE MORANGO                3
VINHO CASILLERO DEL DIABLO GARRAFA      2
VIENNA LITRO/ GROWLER             

In [33]:
tabela_segmentada['descricao'].value_counts()
tabela_segmentada.replace({'COCADA DE FORNO': 'sobremesa',
 'TORTA CRUMBLE DE MORANGO': 'sobremesa',
 'PUDIM TRADICIONAL': 'sobremesa'}, inplace=True)
tabela_segmentada
tabela_segmentada['descricao'].value_counts()

alimento                              666
chopp                                 288
sobremesa                              52
AGUA                                   45
AGUA COM GAS                           44
COCA LATA                              41
GUARANA LATA                           40
UVA 300ML                              39
LARANJA 300ML                          35
TAÇA DE ESPUMANTE                      33
LARANJA 900ML                          32
UVA 900ML                              27
HEINEKEN SEM ALCOOL                    26
CACHAÇA                                25
PAIOL UNIDADE                          12
IPA LITRO/ GROWLER                      8
PILSEN LITRO/ GROWLER                   6
CHOPP VINHO LITRO/ GROWLER              6
BELGA LITRO/ GROWLER                    6
UVA                                     3
VIENNA LITRO/ GROWLER                   2
VINHO CASILLERO DEL DIABLO GARRAFA      2
VINHO SANTA HELENA GARRAFA              1
SESSION IPA                       

In [34]:
tabela_segmentada['descricao'].value_counts()
tabela_segmentada.replace({'AGUA': 'bebidas',
 'AGUA COM GAS': 'bebidas',
 'CACHAÇA': 'bebidas',
 'COCADA DE FORNO': 'bebidas',
 'COCA LATA': 'bebidas',
 'GUARANA LATA': 'bebidas',
 'HEINEKEN SEM ALCOOL': 'bebidas',
 'LARANJA 300ML': 'bebidas',
 'LARANJA 900ML': 'bebidas',
 'TAÇA DE ESPUMANTE': 'bebidas',
 'UVA 300ML': 'bebidas',
 'IPA LITRO/ GROWLER': 'bebidas',
 'UVA 900ML': 'bebidas',
 'VINHO CASILLERO DEL DIABLO GARRAFA': 'bebidas',
 'PILSEN LITRO/ GROWLER': 'bebidas',
 'CHOPP VINHO LITRO/ GROWLER': 'bebidas',
 'BELGA LITRO/ GROWLER': 'bebidas',
 'UVA': 'bebidas',
 'VINHO SANTA HELENA GARRAFA': 'bebidas',
 'PUDIM TRADICIONAL': 'bebidas',
 'VIENNA LITRO/ GROWLER': 'bebidas',
 'SESSION IPA': 'bebidas'}, inplace=True)
tabela_segmentada
tabela_segmentada['descricao'].value_counts()

alimento         666
bebidas          422
chopp            288
sobremesa         52
PAIOL UNIDADE     12
MAÇO PAIOL         1
Name: descricao, dtype: int64

In [35]:
tabela_segmentada['descricao'].value_counts()
tabela_segmentada.replace({'MOLHO MAIONESE',
'MOLHO PIMENTAO VERMELHO',
'MOLHO CHIMICHURI', 
'MOLHO MOSTARDA DIJON'    }, inplace=True)
tabela_segmentada
tabela_segmentada['descricao'].value_counts()

alimento         666
bebidas          422
chopp            288
sobremesa         52
PAIOL UNIDADE     12
MAÇO PAIOL         1
Name: descricao, dtype: int64

In [36]:
tabela_segmentada.drop(tabela_segmentada.loc[tabela_segmentada['descricao'] == 'PAIOL UNIDADE'].index, inplace=True)
tabela_segmentada

cod descricao  quantidade_vendida  valor_unitario  valor_bruto  \
0      72   bebidas                  11             4.0         44.0   
1      73   bebidas                   6             5.0         30.0   
2     362  alimento                   4            22.0         88.0   
3      42  alimento                   5            28.0        140.0   
4      83  alimento                   3            40.0        120.0   
...   ...       ...                 ...             ...          ...   
1436   48  alimento                   4           100.0        400.0   
1437  231  alimento                  12            55.0        660.0   
1438   68   bebidas                   6             6.0         36.0   
1439   70   bebidas                   1            13.0         13.0   
1440   63     chopp                  21            13.0        273.0   

           data  
0    2022-01-01  
1    2022-01-01  
2    2022-01-01  
3    2022-01-01  
4    2022-01-01  
...         ...  
1436 2022-04-17  
1437 2022-04-17  
1438 2022-04-17  
1439 2022-04-17  
1440 2022-04-17  

[1429 rows x 6 columns]

In [37]:
tabela_segmentada.drop(tabela_segmentada.loc[tabela_segmentada['descricao'] == 'MAÇO PAIOL'].index, inplace=True)
tabela_segmentada

cod descricao  quantidade_vendida  valor_unitario  valor_bruto  \
0      72   bebidas                  11             4.0         44.0   
1      73   bebidas                   6             5.0         30.0   
2     362  alimento                   4            22.0         88.0   
3      42  alimento                   5            28.0        140.0   
4      83  alimento                   3            40.0        120.0   
...   ...       ...                 ...             ...          ...   
1436   48  alimento                   4           100.0        400.0   
1437  231  alimento                  12            55.0        660.0   
1438   68   bebidas                   6             6.0         36.0   
1439   70   bebidas                   1            13.0         13.0   
1440   63     chopp                  21            13.0        273.0   

           data  
0    2022-01-01  
1    2022-01-01  
2    2022-01-01  
3    2022-01-01  
4    2022-01-01  
...         ...  
1436 2022-04-17  
1437 2022-04-17  
1438 2022-04-17  
1439 2022-04-17  
1440 2022-04-17  

[1428 rows x 6 columns]

In [38]:
tabela_segmentada2 = tabela_segmentada[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto']].groupby(['descricao']).sum()
tabela_segmentada2

cod  quantidade_vendida  valor_bruto
descricao                                        
alimento   84940                2461      85655.5
bebidas    47639                2199      15196.0
chopp      22525                6161      81895.0
sobremesa  15205                 264       4576.0

In [39]:
tabela_segmentada2

cod  quantidade_vendida  valor_bruto
descricao                                        
alimento   84940                2461      85655.5
bebidas    47639                2199      15196.0
chopp      22525                6161      81895.0
sobremesa  15205                 264       4576.0

In [40]:
tabela_segmentada2.to_excel('tabela_segmentada.xlsx', index=True)

# Faturamento nos meses
Lembrando que a análise começou no dia 01/01/2022 e terminou no dia 17/04/2022, o mês de abril está comprometido e o representa parcialmente. 

##Vendas em Janeiro

In [41]:
tabela_vendas['data'].dt.month_name()

meses_ptbr = {'January':'Janeiro', 'February':'Fevereiro', 'March':'Março', 'April':'Abril', 'May':'Maio', 'June':'Junho', 'July':'Julho', 'August':'Agosto', 'September':'Setembro', 'October':'Outubro', 'November':'Novembro', 'December':'Dezembro'}
tabela_vendas['data'] = tabela_vendas['data'].dt.month_name().replace(meses_ptbr)

In [42]:
vendas_janeiro = tabela_vendas['data'] == 'Janeiro'
vendas_janeiro = tabela_vendas[vendas_janeiro]
vendas_janeiro

cod                 descricao  quantidade_vendida  valor_unitario  \
0     72                      AGUA                  11             4.0   
1     73              AGUA COM GAS                   6             5.0   
2    362  BATATA FRITA TRADICIONAL                   4            22.0   
3     42            BATATA RUSTICA                   5            28.0   
4     83       BOLINHO DE BACALHAU                   3            40.0   
..   ...                       ...                 ...             ...   
471   60                    PILSEN                  91            11.0   
472   45             PORCO NA LATA                   2            55.0   
473  361         TAÇA DE ESPUMANTE                   1            10.0   
474  231                   TILÁPIA                   5            55.0   
475   68                 UVA 300ML                   1             6.0   

     valor_bruto     data  
0           44.0  Janeiro  
1           30.0  Janeiro  
2           88.0  Janeiro  
3          140.0  Janeiro  
4          120.0  Janeiro  
..           ...      ...  
471       1001.0  Janeiro  
472        110.0  Janeiro  
473         10.0  Janeiro  
474        275.0  Janeiro  
475          6.0  Janeiro  

[476 rows x 6 columns]

In [43]:
item_janeiro = vendas_janeiro[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto', 'data']].groupby(['cod', 'descricao', 'data']).sum()
item_janeiro

quantidade_vendida  \
cod descricao                           data                          
40  CONSERVA SPERANZA                   Janeiro                  17   
41  DADINHO DE TAPIOCA                  Janeiro                  47   
42  BATATA RUSTICA                      Janeiro                  10   
44  LINGUIÇA                            Janeiro                  42   
45  PORCO NA LATA                       Janeiro                  35   
46  ISCA CARNE SOL                      Janeiro                  31   
48  STEAK ANCHO                         Janeiro                  19   
50  FAROFA BACON                        Janeiro                  11   
51  CONSERVAS PEPINO                    Janeiro                   4   
52  MOLHO MAIONESE                      Janeiro                  15   
53  ARROZ                               Janeiro                  56   
54  FEIJÃO                              Janeiro                  22   
57  REGUA DEGUSTACAO                    Janeiro                  92   
59  ADICIONAL PAO                       Janeiro                   3   
60  PILSEN                              Janeiro                 955   
62  IPA                                 Janeiro                 415   
63  VIENNA LAGER                        Janeiro                 242   
64  CHOPP VINHO                         Janeiro                 156   
65  GIN                                 Janeiro                  41   
66  COCA LATA                           Janeiro                  98   
67  GUARANA LATA                        Janeiro                  50   
68  UVA 300ML                           Janeiro                  50   
69  LARANJA 300ML                       Janeiro                  40   
70  UVA 900ML                           Janeiro                  26   
71  LARANJA 900ML                       Janeiro                  31   
72  AGUA                                Janeiro                 127   
73  AGUA COM GAS                        Janeiro                  71   
75  CACHAÇA                             Janeiro                  25   
76  MOLHO CHIMICHURI                    Janeiro                   2   
77  MOLHO MOSTARDA DIJON                Janeiro                   1   
78  MOLHO PIMENTAO VERMELHO             Janeiro                   2   
79  CONSERVA CEBOLA                     Janeiro                   1   
80  CONSERVA GILO                       Janeiro                   4   
81  PICANHA                             Janeiro                  18   
82  CORDEIRO PERNIL                     Janeiro                  12   
83  BOLINHO DE BACALHAU                 Janeiro                  28   
85  FRITAS TRADICIONAL                  Janeiro                  26   
86  HEINEKEN SEM ALCOOL                 Janeiro                  18   
206 BELGA                               Janeiro                  66   
212 PRATO KID'S                         Janeiro                  27   
231 TILÁPIA                             Janeiro                  86   
261 MEIA PORÇÃO DE PICANHA              Janeiro                   2   
266 MEIA PORÇÃO DE PORCO NA LATA        Janeiro                   2   
267 MEIA PORÇÃO DE ISCA DE CARNE DE SOL Janeiro                   1   
269 MEIA PORÇÃO DE FRITAS               Janeiro                   5   
298 VINHO CASILLERO DEL DIABLO GARRAFA  Janeiro                   2   
322 PILSEN LITRO/ GROWLER               Janeiro                   1   
325 IPA LITRO/ GROWLER                  Janeiro                   3   
331 RAGÚ DE CUPIM                       Janeiro                  30   
334 PURÊ DE BANANA DA TERRA             Janeiro                   6   
345 COSTELA DE PORCO DEFUMADA           Janeiro                   8   
355 COCADA DE FORNO                     Janeiro                  90   
357 PAIOL UNIDADE                       Janeiro                   1   
361 TAÇA DE ESPUMANTE                   Janeiro                  28   
362 BATATA FRITA TRADICIONAL            Janeiro                  59   
37

In [44]:
venda_total_janeiro = item_janeiro['valor_bruto'].sum()
print(f'Em Janeiro, o faturamento bruto foi de R$ {venda_total_janeiro}')

Em Janeiro, o faturamento bruto foi de R$ 55768.7


No mês de Janeiro, foi constatado um faturamento de R$55768.7

## Vendas Fevereiro

In [45]:
vendas_fevereiro = tabela_vendas['data'] == 'Fevereiro'
vendas_fevereiro = tabela_vendas[vendas_fevereiro]
vendas_fevereiro

cod            descricao  quantidade_vendida  valor_unitario  \
476   72                 AGUA                   1             4.0   
477   73         AGUA COM GAS                   1             5.0   
478  206                BELGA                   2            16.0   
479   83  BOLINHO DE BACALHAU                   1            40.0   
480   64          CHOPP VINHO                   2            12.0   
..   ...                  ...                 ...             ...   
846  361    TAÇA DE ESPUMANTE                   6            10.0   
847  231              TILÁPIA                  15            55.0   
848   68            UVA 300ML                   5             6.0   
849   70            UVA 900ML                   6            13.0   
850   63         VIENNA LAGER                  15            13.0   

     valor_bruto       data  
476          4.0  Fevereiro  
477          5.0  Fevereiro  
478         32.0  Fevereiro  
479         40.0  Fevereiro  
480         24.0  Fevereiro  
..           ...        ...  
846         60.0  Fevereiro  
847        825.0  Fevereiro  
848         30.0  Fevereiro  
849         78.0  Fevereiro  
850        195.0  Fevereiro  

[375 rows x 6 columns]

In [46]:
item_fevereiro = vendas_fevereiro[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto', 'data']].groupby(['cod', 'descricao', 'data']).sum()
item_fevereiro

quantidade_vendida  \
cod descricao                           data                            
40  CONSERVA SPERANZA                   Fevereiro                  23   
41  DADINHO DE TAPIOCA                  Fevereiro                  40   
44  LINGUIÇA                            Fevereiro                  50   
45  PORCO NA LATA                       Fevereiro                  31   
46  ISCA CARNE SOL                      Fevereiro                  32   
48  STEAK ANCHO                         Fevereiro                  31   
50  FAROFA BACON                        Fevereiro                  14   
52  MOLHO MAIONESE                      Fevereiro                   8   
53  ARROZ                               Fevereiro                  68   
54  FEIJÃO                              Fevereiro                  31   
57  REGUA DEGUSTACAO                    Fevereiro                  66   
59  ADICIONAL PAO                       Fevereiro                   7   
60  PILSEN                              Fevereiro                 767   
62  IPA                                 Fevereiro                 244   
63  VIENNA LAGER                        Fevereiro                 168   
64  CHOPP VINHO                         Fevereiro                 104   
65  GIN                                 Fevereiro                  32   
66  COCA LATA                           Fevereiro                 111   
67  GUARANA LATA                        Fevereiro                  40   
68  UVA 300ML                           Fevereiro                  31   
69  LARANJA 300ML                       Fevereiro                  43   
70  UVA 900ML                           Fevereiro                  27   
71  LARANJA 900ML                       Fevereiro                  23   
72  AGUA                                Fevereiro                 106   
73  AGUA COM GAS                        Fevereiro                  75   
75  CACHAÇA                             Fevereiro                  22   
76  MOLHO CHIMICHURI                    Fevereiro                   3   
78  MOLHO PIMENTAO VERMELHO             Fevereiro                   8   
80  CONSERVA GILO                       Fevereiro                   2   
81  PICANHA                             Fevereiro                  26   
82  CORDEIRO PERNIL                     Fevereiro                  10   
83  BOLINHO DE BACALHAU                 Fevereiro                  30   
86  HEINEKEN SEM ALCOOL                 Fevereiro                  21   
110 UVA                                 Fevereiro                   1   
206 BELGA                               Fevereiro                 133   
212 PRATO KID'S                         Fevereiro                  16   
231 TILÁPIA                             Fevereiro                  88   
241 SESSION IPA                         Fevereiro                   1   
261 MEIA PORÇÃO DE PICANHA              Fevereiro                   2   
266 MEIA PORÇÃO DE PORCO NA LATA        Fevereiro                   1   
267 MEIA PORÇÃO DE ISCA DE CARNE DE SOL Fevereiro                   2   
269 MEIA PORÇÃO DE FRITAS               Fevereiro                   1   
297 VINHO SANTA HELENA GARRAFA          Fevereiro                   1   
298 VINHO CASILLERO DEL DIABLO GARRAFA  Fevereiro                   1   
322 PILSEN LITRO/ GROWLER               Fevereiro                   3   
323 BELGA LITRO/ GROWLER                Fevereiro                   3   
325 IPA LITRO/ GROWLER                  Fevereiro                  12   
327 CHOPP VINHO LITRO/ GROWLER          Fevereiro                   3   
331 RAGÚ DE CUPIM                       Fevereiro                  13   
334 PURÊ DE BANANA DA TERRA             Fevereiro                  14   
355 COCADA DE FORNO                     Fevereiro                  81   
357 PAIOL UNIDADE                       Fevereiro                   6   
361 TAÇA DE ESPUMANTE                   Fevereiro                  25   
362 BATATA FRITA TRADICIONAL        

In [47]:
venda_total_fevereiro = item_fevereiro['valor_bruto'].sum()
print(f'Em Fevereiro, o faturamento bruto foi de R$ {venda_total_fevereiro}')

Em Fevereiro, o faturamento bruto foi de R$ 50090.3


No mês de Fevereiro, foi constatado um faturamento de R$ 50090.30

## Vendas Março

In [48]:
vendas_marco = tabela_vendas['data'] == 'Março'
vendas_marco = tabela_vendas[vendas_marco]
vendas_marco

cod                 descricao  quantidade_vendida  valor_unitario  \
851    59             ADICIONAL PAO                   1             5.9   
852    72                      AGUA                   9             4.0   
853    73              AGUA COM GAS                   9             5.0   
854    53                     ARROZ                   8             6.0   
855   362  BATATA FRITA TRADICIONAL                  12            22.0   
...   ...                       ...                 ...             ...   
1187  361         TAÇA DE ESPUMANTE                   5            10.0   
1188  231                   TILÁPIA                   9            55.0   
1189   68                 UVA 300ML                   7             6.0   
1190   70                 UVA 900ML                   2            13.0   
1191   63              VIENNA LAGER                  22            13.0   

      valor_bruto   data  
851           5.9  Março  
852          36.0  Março  
853          45.0  Março  
854          48.0  Março  
855         264.0  Março  
...           ...    ...  
1187         50.0  Março  
1188        495.0  Março  
1189         42.0  Março  
1190         26.0  Março  
1191        286.0  Março  

[341 rows x 6 columns]

In [49]:
item_marco = vendas_marco[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto', 'data']].groupby(['cod', 'descricao', 'data']).sum()
item_marco

quantidade_vendida  valor_bruto
cod descricao             data                                  
40  CONSERVA SPERANZA     Março                  12        312.0
41  DADINHO DE TAPIOCA    Março                  27        864.0
44  LINGUIÇA              Março                  29       1305.0
45  PORCO NA LATA         Março                  10        550.0
46  ISCA CARNE SOL        Março                  23       1150.0
...                                             ...          ...
393 EVENTO ISCA DE FRANGO Março                  31        775.0
394 EVENTO ISCA DE PORCO  Março                  39        975.0
395 EVENTO BATATA FRITA   Março                 159       3180.0
396 ARROZ CARRETEIRO      Março                 112       1755.0
397 EVENTO CHOPP VERDE    Março                  62        744.0

[61 rows x 2 columns]

In [50]:
venda_total_marco = item_marco['valor_bruto'].sum()
print(f'Em Março, o faturamento bruto foi de R$ {venda_total_marco}')

Em Março, o faturamento bruto foi de R$ 49707.5


No mês de Março, foi constatado um faturamento de R$ 49707.5

## Vendas Abril 

In [51]:
vendas_abril = tabela_vendas['data'] == 'Abril'
vendas_abril = tabela_vendas[vendas_abril]
vendas_abril

cod                 descricao  quantidade_vendida  valor_unitario  \
1192   72                      AGUA                   1             4.0   
1193   73              AGUA COM GAS                   6             5.0   
1194  362  BATATA FRITA TRADICIONAL                   3            22.0   
1195   83       BOLINHO DE BACALHAU                   2            40.0   
1196   64               CHOPP VINHO                   1            12.0   
...   ...                       ...                 ...             ...   
1436   48               STEAK ANCHO                   4           100.0   
1437  231                   TILÁPIA                  12            55.0   
1438   68                 UVA 300ML                   6             6.0   
1439   70                 UVA 900ML                   1            13.0   
1440   63              VIENNA LAGER                  21            13.0   

      valor_bruto   data  
1192          4.0  Abril  
1193         30.0  Abril  
1194         66.0  Abril  
1195         80.0  Abril  
1196         12.0  Abril  
...           ...    ...  
1436        400.0  Abril  
1437        660.0  Abril  
1438         36.0  Abril  
1439         13.0  Abril  
1440        273.0  Abril  

[249 rows x 6 columns]

In [52]:
item_abril = vendas_abril[['cod', 'descricao', 'quantidade_vendida', 'valor_bruto', 'data']].groupby(['cod', 'descricao', 'data']).sum()
item_abril

quantidade_vendida  valor_bruto
cod descricao                           data                                  
40  CONSERVA SPERANZA                   Abril                  12        312.0
41  DADINHO DE TAPIOCA                  Abril                  23        736.0
44  LINGUIÇA                            Abril                  18        810.0
45  PORCO NA LATA                       Abril                  26       1430.0
46  ISCA CARNE SOL                      Abril                  14        700.0
48  STEAK ANCHO                         Abril                  15       1500.0
50  FAROFA BACON                        Abril                   7         42.0
51  CONSERVAS PEPINO                    Abril                   4         24.0
52  MOLHO MAIONESE                      Abril                   9         54.0
53  ARROZ                               Abril                  34        204.0
54  FEIJÃO                              Abril                   8         48.0
55  PUDIM TRADICIONAL                   Abril                  13        130.0
57  REGUA DEGUSTACAO                    Abril                  31        868.0
59  ADICIONAL PAO                       Abril                   1          6.0
60  PILSEN                              Abril                 610       6710.0
62  IPA                                 Abril                 201       2814.0
63  VIENNA LAGER                        Abril                  90       1170.0
64  CHOPP VINHO                         Abril                  96       1152.0
65  GIN                                 Abril                  24        600.0
66  COCA LATA                           Abril                  90        540.0
67  GUARANA LATA                        Abril                  27        162.0
68  UVA 300ML                           Abril                  22        132.0
69  LARANJA 300ML                       Abril                  25        150.0
70  UVA 900ML                           Abril                   7         91.0
71  LARANJA 900ML                       Abril                   5         65.0
72  AGUA                                Abril                  68        272.0
73  AGUA COM GAS                        Abril                  55        275.0
75  CACHAÇA                             Abril                  10         60.0
76  MOLHO CHIMICHURI                    Abril                   1          6.0
78  MOLHO PIMENTAO VERMELHO             Abril                   3         18.0
79  CONSERVA CEBOLA                     Abril                   1          6.0
80  CONSERVA GILO                       Abril                   3         18.0
81  PICANHA                             Abril                  19       2090.0
82  CORDEIRO PERNIL                     Abril                   4        360.0
83  BOLINHO DE BACALHAU                 Abril                  23        920.0
86  HEINEKEN SEM ALCOOL                 Abril                  11        110.0
206 BELGA                               Abril                  64       1024.0
212 PRATO KID'S                         Abril                  12        360.0
231 TILÁPIA                             Abril                  48       2640.0
265 MEIA PORÇÃO DE LINGUIÇA             Abril                   1         27.0
267 MEIA PORÇÃO DE ISCA DE CARNE DE SOL Abril                   1         30.0
323 BELGA LITRO/ GROWLER                Abril                   2         72.0
326 VIENNA LITRO/ GROWLER               Abril                   2         68.0
327 CHOPP VINHO LITRO/ GROWLER          Abril                   1         34.0
334 PURÊ DE BANANA DA TERRA             Abril                   5         50.0
355 COCADA DE FORNO                     Abril                  35        630.0
357 PAIOL UNIDADE                       Abril                   4          8.0
361 TAÇA DE ESPUMANTE                   Abril                  11        110.0
362 BATATA FRITA TRADICIONAL            Abril                  58       1276.0
396 ARROZ CARRETEI

In [53]:
venda_total_abril = item_abril['valor_bruto'].sum()
print(f'Em Abril, o faturamento bruto foi de R$ {venda_total_abril}')

Em Abril, o faturamento bruto foi de R$ 31814.0


No mês de Abril, o faturamento bruto até o dia 17 foi de R$ 31814.0

# Vendas por dia da semana
Para ter uma visão mais realista, vamos destrinchar a tabela em dias para podermos comparar o faturamento de cada dia da semana.

In [54]:
tabela_dias = pd.read_excel('Analise_2022_SPZ.xlsx')
tabela_dias

renomear = {'Cód produto' : 'cod',
            'Descrição' : 'descricao',
            'Quantidade Vendida' : 'quantidade_vendida',
            'Valor Unitário' : 'valor_unitario',
            'Valor Bruto' : 'valor_bruto',
            'Data' : 'data'}
tabela_dias = tabela_dias.rename(columns = renomear)
tabela_dias.head(5)

cod                 descricao  quantidade_vendida  valor_unitario  \
0   72                      AGUA                  11             4.0   
1   73              AGUA COM GAS                   6             5.0   
2  362  BATATA FRITA TRADICIONAL                   4            22.0   
3   42            BATATA RUSTICA                   5            28.0   
4   83       BOLINHO DE BACALHAU                   3            40.0   

   valor_bruto       data  
0         44.0 2022-01-01  
1         30.0 2022-01-01  
2         88.0 2022-01-01  
3        140.0 2022-01-01  
4        120.0 2022-01-01

In [55]:
tabela_dias['data'].dt.day_name()


0       Saturday
1       Saturday
2       Saturday
3       Saturday
4       Saturday
          ...   
1436      Sunday
1437      Sunday
1438      Sunday
1439      Sunday
1440      Sunday
Name: data, Length: 1441, dtype: object

In [56]:
dias_ptbr = {'Sunday':'Domingo', 'Monday':'Segunda-feira', 'Tuesday':'Terça-feira', 'Wednesday':'Quarta-feira', 'Thursday':'Quinta-feira', 'Friday':'Sexta-feira', 'Saturday': 'Sábado'}
tabela_dias['data'] = tabela_dias['data'].dt.day_name().replace(dias_ptbr)
tabela_dias

cod                 descricao  quantidade_vendida  valor_unitario  \
0      72                      AGUA                  11             4.0   
1      73              AGUA COM GAS                   6             5.0   
2     362  BATATA FRITA TRADICIONAL                   4            22.0   
3      42            BATATA RUSTICA                   5            28.0   
4      83       BOLINHO DE BACALHAU                   3            40.0   
...   ...                       ...                 ...             ...   
1436   48               STEAK ANCHO                   4           100.0   
1437  231                   TILÁPIA                  12            55.0   
1438   68                 UVA 300ML                   6             6.0   
1439   70                 UVA 900ML                   1            13.0   
1440   63              VIENNA LAGER                  21            13.0   

      valor_bruto     data  
0            44.0   Sábado  
1            30.0   Sábado  
2            88.0   Sábado  
3           140.0   Sábado  
4           120.0   Sábado  
...           ...      ...  
1436        400.0  Domingo  
1437        660.0  Domingo  
1438         36.0  Domingo  
1439         13.0  Domingo  
1440        273.0  Domingo  

[1441 rows x 6 columns]

## Vendas Sexta-feira

In [57]:
vendas_sexta = tabela_dias['data'] == 'Sexta-feira'
vendas_sexta = tabela_dias[vendas_sexta]
vendas_sexta

cod                 descricao  quantidade_vendida  valor_unitario  \
93     72                      AGUA                  13             4.0   
94     73              AGUA COM GAS                   4             5.0   
95     53                     ARROZ                   1             6.0   
96    362  BATATA FRITA TRADICIONAL                   2            22.0   
97     42            BATATA RUSTICA                   1            28.0   
...   ...                       ...                 ...             ...   
1284   60                    PILSEN                   7            11.0   
1285   45             PORCO NA LATA                   2            55.0   
1286   48               STEAK ANCHO                   1           100.0   
1287  231                   TILÁPIA                   1            55.0   
1288   63              VIENNA LAGER                   3            13.0   

      valor_bruto         data  
93           52.0  Sexta-feira  
94           20.0  Sexta-feira  
95            6.0  Sexta-feira  
96           44.0  Sexta-feira  
97           28.0  Sexta-feira  
...           ...          ...  
1284         77.0  Sexta-feira  
1285        110.0  Sexta-feira  
1286        100.0  Sexta-feira  
1287         55.0  Sexta-feira  
1288         39.0  Sexta-feira  

[222 rows x 6 columns]

In [58]:
vendas_sexta = vendas_sexta['valor_bruto'].sum()
print(f'Nas sextas-feiras, as vendas totalizaram R$ {vendas_sexta}')

Nas sextas-feiras, as vendas totalizaram R$ 14938.9


## Vendas Sábado

In [59]:
vendas_sabado = tabela_dias['data'] == 'Sábado'
vendas_sabado = tabela_dias[vendas_sabado]
vendas_sabado

cod                 descricao  quantidade_vendida  valor_unitario  \
0      72                      AGUA                  11             4.0   
1      73              AGUA COM GAS                   6             5.0   
2     362  BATATA FRITA TRADICIONAL                   4            22.0   
3      42            BATATA RUSTICA                   5            28.0   
4      83       BOLINHO DE BACALHAU                   3            40.0   
...   ...                       ...                 ...             ...   
1401   48               STEAK ANCHO                   2           100.0   
1402  361         TAÇA DE ESPUMANTE                   3            10.0   
1403  231                   TILÁPIA                   9            55.0   
1404   68                 UVA 300ML                   9             6.0   
1405   63              VIENNA LAGER                  30            13.0   

      valor_bruto    data  
0            44.0  Sábado  
1            30.0  Sábado  
2            88.0  Sábado  
3           140.0  Sábado  
4           120.0  Sábado  
...           ...     ...  
1401        200.0  Sábado  
1402         30.0  Sábado  
1403        495.0  Sábado  
1404         54.0  Sábado  
1405        390.0  Sábado  

[542 rows x 6 columns]

In [60]:
vendas_sabado = vendas_sabado['valor_bruto'].sum()
print(f'Nos sábados, as vendas totalizaram R$ {vendas_sabado}')

Nos sábados, as vendas totalizaram R$ 74694.6


## Vendas Domingo

In [61]:
vendas_domingo = tabela_dias['data'] == 'Domingo'
vendas_domingo = tabela_dias[vendas_domingo]
vendas_domingo

cod                 descricao  quantidade_vendida  valor_unitario  \
31     59             ADICIONAL PAO                   1             5.9   
32     72                      AGUA                  19             4.0   
33     73              AGUA COM GAS                  10             5.0   
34     53                     ARROZ                   6             6.0   
35    362  BATATA FRITA TRADICIONAL                   2            22.0   
...   ...                       ...                 ...             ...   
1436   48               STEAK ANCHO                   4           100.0   
1437  231                   TILÁPIA                  12            55.0   
1438   68                 UVA 300ML                   6             6.0   
1439   70                 UVA 900ML                   1            13.0   
1440   63              VIENNA LAGER                  21            13.0   

      valor_bruto     data  
31            5.9  Domingo  
32           76.0  Domingo  
33           50.0  Domingo  
34           36.0  Domingo  
35           44.0  Domingo  
...           ...      ...  
1436        400.0  Domingo  
1437        660.0  Domingo  
1438         36.0  Domingo  
1439         13.0  Domingo  
1440        273.0  Domingo  

[560 rows x 6 columns]

In [62]:
vendas_domingo = vendas_domingo['valor_bruto'].sum()
print(f'Nos domingos, as vendas totalizaram R$ {vendas_domingo}')

Nos domingos, as vendas totalizaram R$ 83909.5


#Vendas total por semana
Nesse estudo vamos, de modo geral, fazer um somatório do faturamento pelos dias da semana, para tirarmos à média de vendas. 

In [63]:
tabela_dias_mes = pd.read_excel('Analise_2022_SPZ.xlsx')
tabela_dias_mes

Cód produto                 Descrição  Quantidade Vendida  \
0              72                      AGUA                  11   
1              73              AGUA COM GAS                   6   
2             362  BATATA FRITA TRADICIONAL                   4   
3              42            BATATA RUSTICA                   5   
4              83       BOLINHO DE BACALHAU                   3   
...           ...                       ...                 ...   
1436           48               STEAK ANCHO                   4   
1437          231                   TILÁPIA                  12   
1438           68                 UVA 300ML                   6   
1439           70                 UVA 900ML                   1   
1440           63              VIENNA LAGER                  21   

      Valor Unitário  Valor Bruto       Data  
0                4.0         44.0 2022-01-01  
1                5.0         30.0 2022-01-01  
2               22.0         88.0 2022-01-01  
3               28.0        140.0 2022-01-01  
4               40.0        120.0 2022-01-01  
...              ...          ...        ...  
1436           100.0        400.0 2022-04-17  
1437            55.0        660.0 2022-04-17  
1438             6.0         36.0 2022-04-17  
1439            13.0         13.0 2022-04-17  
1440            13.0        273.0 2022-04-17  

[1441 rows x 6 columns]

In [64]:
renomear = {'Cód produto' : 'cod',
            'Descrição' : 'descricao',
            'Quantidade Vendida' : 'quantidade_vendida',
            'Valor Unitário' : 'valor_unitario',
            'Valor Bruto' : 'valor_bruto',
            'Data' : 'data'}
tabela_dias_mes = tabela_dias_mes.rename(columns = renomear)
tabela_dias_mes.head(5)

cod                 descricao  quantidade_vendida  valor_unitario  \
0   72                      AGUA                  11             4.0   
1   73              AGUA COM GAS                   6             5.0   
2  362  BATATA FRITA TRADICIONAL                   4            22.0   
3   42            BATATA RUSTICA                   5            28.0   
4   83       BOLINHO DE BACALHAU                   3            40.0   

   valor_bruto       data  
0         44.0 2022-01-01  
1         30.0 2022-01-01  
2         88.0 2022-01-01  
3        140.0 2022-01-01  
4        120.0 2022-01-01

In [65]:
tabela_dias_group = tabela_dias_mes[['quantidade_vendida', 'valor_bruto', 'data']].groupby('data').sum()
tabela_dias_group

quantidade_vendida  valor_bruto
data                                       
2022-01-01                 271       4748.0
2022-01-02                 402       6947.9
2022-01-06                 109       1817.0
2022-01-07                 307       4973.0
2022-01-08                 339       5461.0
2022-01-09                 197       3432.8
2022-01-13                  43        680.0
2022-01-14                  62       1136.0
2022-01-15                 222       4100.0
2022-01-16                 191       3201.0
2022-01-21                  84       1209.0
2022-01-22                 335       5770.0
2022-01-23                 281       5253.0
2022-01-27                   3         69.0
2022-01-28                  62        869.0
2022-01-29                 150       2816.0
2022-01-30                 204       3286.0
2022-02-04                  26        502.0
2022-02-05                 229       3884.0
2022-02-06                 296       5329.0
2022-02-11                  14        246.0
2022-02-12                 191       3441.0
2022-02-13                 429       7451.9
2022-02-18                  17        439.0
2022-02-19                 262       4673.9
2022-02-20                 330       6330.0
2022-02-25                  74       1359.0
2022-02-26                 235       4190.0
2022-02-27                 305       5766.9
2022-02-28                 372       6477.6
2022-03-01                 297       4793.9
2022-03-04                  88       1550.9
2022-03-05                 233       3741.0
2022-03-06                 210       3270.0
2022-03-11                  56        808.0
2022-03-12                 250       4040.8
2022-03-13                 193       3291.0
2022-03-19                 768      10976.0
2022-03-20                 517       7291.0
2022-03-25                  19        401.0
2022-03-26                 159       2782.9
2022-03-27                 393       6761.0
2022-04-01                  53        853.0
2022-04-02                 239       3618.0
2022-04-03                 237       4598.0
2022-04-08                  32        593.0
2022-04-09                 208       3466.0
2022-04-10                 376       6095.0
2022-04-16                 430       6986.0
2022-04-17                 305       5605.0

In [66]:
tabela_dias_group.to_excel('tabela_dias_mes.xlsx', index=True)

In [67]:
tabela_dias_group = tabela_dias_group.reset_index()
tabela_dias_group

data  quantidade_vendida  valor_bruto
0  2022-01-01                 271       4748.0
1  2022-01-02                 402       6947.9
2  2022-01-06                 109       1817.0
3  2022-01-07                 307       4973.0
4  2022-01-08                 339       5461.0
5  2022-01-09                 197       3432.8
6  2022-01-13                  43        680.0
7  2022-01-14                  62       1136.0
8  2022-01-15                 222       4100.0
9  2022-01-16                 191       3201.0
10 2022-01-21                  84       1209.0
11 2022-01-22                 335       5770.0
12 2022-01-23                 281       5253.0
13 2022-01-27                   3         69.0
14 2022-01-28                  62        869.0
15 2022-01-29                 150       2816.0
16 2022-01-30                 204       3286.0
17 2022-02-04                  26        502.0
18 2022-02-05                 229       3884.0
19 2022-02-06                 296       5329.0
20 2022-02-11                  14        246.0
21 2022-02-12                 191       3441.0
22 2022-02-13                 429       7451.9
23 2022-02-18                  17        439.0
24 2022-02-19                 262       4673.9
25 2022-02-20                 330       6330.0
26 2022-02-25                  74       1359.0
27 2022-02-26                 235       4190.0
28 2022-02-27                 305       5766.9
29 2022-02-28                 372       6477.6
30 2022-03-01                 297       4793.9
31 2022-03-04                  88       1550.9
32 2022-03-05                 233       3741.0
33 2022-03-06                 210       3270.0
34 2022-03-11                  56        808.0
35 2022-03-12                 250       4040.8
36 2022-03-13                 193       3291.0
37 2022-03-19                 768      10976.0
38 2022-03-20                 517       7291.0
39 2022-03-25                  19        401.0
40 2022-03-26                 159       2782.9
41 2022-03-27                 393       6761.0
42 2022-04-01                  53        853.0
43 2022-04-02                 239       3618.0
44 2022-04-03                 237       4598.0
45 2022-04-08                  32        593.0
46 2022-04-09                 208       3466.0
47 2022-04-10                 376       6095.0
48 2022-04-16                 430       6986.0
49 2022-04-17                 305       5605.0

In [68]:
tabela_dias_group['data'] = tabela_dias_group['data'].dt.day_name().replace(dias_ptbr)
tabela_dias_group

data  quantidade_vendida  valor_bruto
0          Sábado                 271       4748.0
1         Domingo                 402       6947.9
2    Quinta-feira                 109       1817.0
3     Sexta-feira                 307       4973.0
4          Sábado                 339       5461.0
5         Domingo                 197       3432.8
6    Quinta-feira                  43        680.0
7     Sexta-feira                  62       1136.0
8          Sábado                 222       4100.0
9         Domingo                 191       3201.0
10    Sexta-feira                  84       1209.0
11         Sábado                 335       5770.0
12        Domingo                 281       5253.0
13   Quinta-feira                   3         69.0
14    Sexta-feira                  62        869.0
15         Sábado                 150       2816.0
16        Domingo                 204       3286.0
17    Sexta-feira                  26        502.0
18         Sábado                 229       3884.0
19        Domingo                 296       5329.0
20    Sexta-feira                  14        246.0
21         Sábado                 191       3441.0
22        Domingo                 429       7451.9
23    Sexta-feira                  17        439.0
24         Sábado                 262       4673.9
25        Domingo                 330       6330.0
26    Sexta-feira                  74       1359.0
27         Sábado                 235       4190.0
28        Domingo                 305       5766.9
29  Segunda-feira                 372       6477.6
30    Terça-feira                 297       4793.9
31    Sexta-feira                  88       1550.9
32         Sábado                 233       3741.0
33        Domingo                 210       3270.0
34    Sexta-feira                  56        808.0
35         Sábado                 250       4040.8
36        Domingo                 193       3291.0
37         Sábado                 768      10976.0
38        Domingo                 517       7291.0
39    Sexta-feira                  19        401.0
40         Sábado                 159       2782.9
41        Domingo                 393       6761.0
42    Sexta-feira                  53        853.0
43         Sábado                 239       3618.0
44        Domingo                 237       4598.0
45    Sexta-feira                  32        593.0
46         Sábado                 208       3466.0
47        Domingo                 376       6095.0
48         Sábado                 430       6986.0
49        Domingo                 305       5605.0

In [69]:
quantidade_de_dias_trabalhados = tabela_dias_group['data'].value_counts()
quantidade_de_dias_trabalhados

Sábado           16
Domingo          16
Sexta-feira      13
Quinta-feira      3
Segunda-feira     1
Terça-feira       1
Name: data, dtype: int64

In [72]:
tabela_dias_semana = tabela_dias_group[['quantidade_vendida', 'valor_bruto', 'data']].groupby('data').sum()
tabela_dias_semana

quantidade_vendida  valor_bruto
data                                          
Domingo                      4866      83909.5
Quinta-feira                  155       2566.0
Segunda-feira                 372       6477.6
Sexta-feira                   894      14938.9
Sábado                       4521      74694.6
Terça-feira                   297       4793.9

In [74]:
tabela_dias_semana['dias_trabalhados'] = quantidade_de_dias_trabalhados
tabela_dias_semana

quantidade_vendida  valor_bruto  dias_trabalhados
data                                                            
Domingo                      4866      83909.5                16
Quinta-feira                  155       2566.0                 3
Segunda-feira                 372       6477.6                 1
Sexta-feira                   894      14938.9                13
Sábado                       4521      74694.6                16
Terça-feira                   297       4793.9                 1

In [77]:
tabela_dias_semana['media_venda'] = tabela_dias_semana['valor_bruto'] / tabela_dias_semana['dias_trabalhados']
tabela_dias_semana.round(2)

quantidade_vendida  valor_bruto  dias_trabalhados  media_venda
data                                                                         
Domingo                      4866      83909.5                16      5244.34
Quinta-feira                  155       2566.0                 3       855.33
Segunda-feira                 372       6477.6                 1      6477.60
Sexta-feira                   894      14938.9                13      1149.15
Sábado                       4521      74694.6                16      4668.41
Terça-feira                   297       4793.9                 1      4793.90

#Análise Sexta-Feira
Ao termos acesso às vendas por dia, nota-se que na sexta feira, há um dia não condizente com a mediana. A exclusão de Outliers será fundamental para termos ciência da realidade. No sábado e domingo não houve necessidade da remoção de Outliers. 

In [ ]:
dias = pd.read_excel('Analise_2022_SPZ.xlsx')
dias

In [ ]:
renomear = {'Cód produto' : 'cod',
            'Descrição' : 'descricao',
            'Quantidade Vendida' : 'quantidade_vendida',
            'Valor Unitário' : 'valor_unitario',
            'Valor Bruto' : 'valor_bruto',
            'Data' : 'data'}
dias = dias.rename(columns = renomear)
dias.head(5)

In [ ]:
dias = tabela_dias_mes[['quantidade_vendida', 'valor_bruto', 'data']].groupby('data').sum()
dias

In [ ]:
dias = dias.reset_index()
dias

In [ ]:
dias['dia_semana'] = dias['data'].dt.day_name().replace(dias_ptbr)
dias

In [ ]:
dias_sexta = dias['dia_semana'] == 'Sexta-feira'
dias_sexta = dias[dias_sexta]
dias_sexta

In [ ]:
dias_sexta.boxplot(column='valor_bruto', figsize=(15,8))
plt.show()

In [ ]:
vari_sexta = dias_sexta['valor_bruto']

In [ ]:
Q1 = vari_sexta.quantile(.25)
Q3 = vari_sexta.quantile(.75)
IIQ = Q3 - Q1
limite_inferior = Q1 - 1.5 * IIQ
limite_superior = Q3 + 1.5 * IIQ

In [ ]:
selecao = (vari_sexta >= limite_inferior) & ( vari_sexta <= limite_superior)
sexta_boxplot = dias_sexta[selecao]
sexta_boxplot

In [ ]:
sexta_boxplot.boxplot(['valor_bruto'], figsize=(15,8))

In [ ]:
sexta_boxplot['valor_bruto'].mean()

In [ ]:
sexta_boxplot

In [ ]:
sexta_boxplot.to_excel('vendas_sexta.xlsx', index=True)

##Analisando a sexta feira, percebo que há um outliers. Na primeira sexta-feira do ano, há um dado que difere da realidade, aumentando a venda média por dia. Removendo-o, percebe-se que há um equilibrio na média de vendas.


# CONCLUSÃO

Após a análise feita, concluímos que há necessidade em estudar alguma possibilidade de mudança na sexta-feira, visto que o faturamento médio é 4x menor que no sábado e 5x menor que no domingo. Partindo do princípio que podemos melhorar o faturamento geral, sugerimos neste estudo:

- Ampliação horário de funcionamento;
- Criação de promoções atrativas;
- Criação de um happy hour;
- Possibilidade de colocar música ao vivo;
- Investir em um ambiente propício para noite.